src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -20,4 +33,28 @@ public String greeting(@RequestParam(name="name", required=false, defaultValue="
     return "greeting";
   }
 
+  @GetMapping("/{id}")
+  public Book read(@PathVariable Long id) throws NotFoundException {
+    return bookRepository.findById(id).orElseThrow(() -> new NotFoundException(notFoundMessage));
+  }
+
+  @PostMapping
+  @ResponseStatus(HttpStatus.CREATED)
+  public Book create(@RequestBody Book book) {
+    return bookRepository.save(book);
+  }
+
+  @DeleteMapping("/{id}")
+  public void delete(@PathVariable Long id) {
+    bookRepository.findById(id).orElseThrow(() -> new NotFoundException(notFoundMessage));
+    bookRepository.deleteById(id);
+  }
+
+  @PutMapping("/{id}")
+  public Book update(@RequestBody Book book, @PathVariable Long id) {
+    bookRepository.findById(id).orElseThrow(() -> new NotFoundException(notFoundMessage));
+    if (!book.getId().equals(id)) throw new ForbiddenException( "Cannot change 'id'");
```

Comment:
It is good practice to have all error messages as constants in a separate class, we should not have them literal in the place where the exception is raised

Reply:
Where should the class be? Under the main package?

Reply:
yes, and under a package named `constants` for more readability